GenderPayGap.WebUI/Controllers/Report/NewReportOverviewController.cs

```diff
@@ -0,0 +1,277 @@
+            ControllerHelper.ThrowIfReportingYearIsOutsideOfRange(reportingYear);
+            
+            DraftReturn draftReturn = draftReturnService.GetDraftReturn(organisationId, reportingYear);
+            
+            if (!draftReturnService.DraftReturnExistsAndRequiredFieldsAreComplete(organisationId, reportingYear))
+            {
+                return RedirectToReportOverviewPage(encryptedOrganisationId, reportingYear, "This report is not ready to submit. Complete the remaining sections");
+            }
+            
+            
+            if (WillBeLateSubmission(draftReturn))
+            {
+                // Late submission Reason
+                return RedirectToAction("NewLateSubmissionReasonGet", "NewLateSubmission", new {encryptedOrganisationId = encryptedOrganisationId, reportingYear = reportingYear});
+            }
+            else
+            {
+                User user = ControllerHelper.GetGpgUserFromAspNetUser(User, dataRepository);
+                Return newReturn = returnService.CreateAndSaveReturnFromDraftReturn(draftReturn, user, Url);
+
+                // Confirmation
+                return RedirectToAction("NewReportConfirmation", "NewReportConfirmation",
+                    new
+                    {
+                        encryptedOrganisationId = encryptedOrganisationId,
+                        reportingYear = reportingYear,
+                        confirmationId = Encryption.EncryptQuerystring(newReturn.ReturnId.ToString())
+                    });
+            }
+        }
+
+        private IActionResult RedirectToReportOverviewPage(string encryptedOrganisationId, int reportingYear, string message)
+        {
+            string nextPageUrl = Url.Action("NewReportOverview", "NewReportOverview", new {encryptedOrganisationId = encryptedOrganisationId, reportingYear = reportingYear});
+            StatusMessageHelper.SetStatusMessage(Response, message, nextPageUrl);
+            return LocalRedirect(nextPageUrl);
+        }
+        
+        private bool WillBeLateSubmission(DraftReturn draftReturn)
+        {
+            return draftReturnService.DraftReturnWouldBeNewlyLateIfSubmittedNow(draftReturn);
+        }
+        
+        [HttpGet("{encryptedOrganisationId}/reporting-year-{reportingYear}/report-new")]
+        public IActionResult NewReportOverview(string encryptedOrganisationId, int reportingYear, bool shouldShowLateSubmissionWarning = false)
+        {
+            long organisationId = ControllerHelper.DecryptOrganisationIdOrThrow404(encryptedOrganisationId);
+            ControllerHelper.ThrowIfUserAccountRetiredOrEmailNotVerified(User, dataRepository);
+            ControllerHelper.ThrowIfUserDoesNotHavePermissionsForGivenOrganisation(User, dataRepository, organisationId);
+            ControllerHelper.ThrowIfReportingYearIsOutsideOfRange(reportingYear);
+
+            if (ReportIsLate(organisationId, reportingYear) && shouldShowLateSubmissionWarning)
```

Comment:
We don't call it because it checks to show the Late submission reason page, not the Late warning page, which is shown when we don't know whether a draft/submitted return exists or not and if they have the neccessary fields changed to require a late reason.

I have refactored this function to use the same functions `DraftReturnWouldBeNewlyLateIfSubmittedNow` and commonised their identical checks and also move this function into the `draftReturnService`